sherpa/astro/tests/test_data.py

```diff
@@ -839,3 +839,23 @@ def test_rmf_with_grid_below_thresh_zero():
     emsg = "The RMF 'delta-rmf' has an ENERG_HI value <= " + \
            "the replacement value of 1e-05"
     assert str(exc.value) == emsg
+
+
+# Bug https://github.com/sherpa/sherpa/issues/572
+@requires_data
+@requires_fits
+def test_arf_rmf_get_x(make_data_path):
```

Comment:
Our main test configurations have all the dependencies, or all the dependencies except xspec and ds9. So as long as tests using xspec and ds9 are the exception and not rule, we are fine. Configurations with a subset of dependencies are ancillary and ensure users with these *special* configurations can still run the tests, not the other way around. So again I think we are fine.

That said, I treated this as a simple regression, so I reproduced the bug in conditions similar to the ones reported originally. Being a regression there is a slight risk in using a unit-like test case (in general, probably not in this case), because there might be subtle differences between data generated using those special methods and data generated from actual files.

Again, I just treated this as a simple regression. Nonetheless, I'll see if I can reproduce the issue with `create_arf` and if it's useful to add a new test using those methods.

Reply:
I think in this case there's a strong argument that reading in the data from a file is a sensible thing to do (in that you essentially need FITS support to do PHA analysis). There is also the fact, as you mention, that for complex data structures like an RMF there may be cases where the "simple" and "from a file" approaches behave differently (but then it would be nice to have tests that reduce the chance that this happens).

I would like to have the "core" functionality easily tested, which is why I "prefer" the tests always run, but I do think it's on a case-by-case basis.